File: PythonDataAnalysis/Impute missing values.R
List_null = data.frame(one=c(1,3,NA,6,8,NA,9),two=c(8,0,1,NA,2,6,NA))
# replace na with column means
List_null[which(!complete.cases(List_null$one)),"one"]=mean(List_null$one,na.rm=T)
List_null[which(is.na(List_null$two)),"two"]=mean(List_null$two,na.rm=T)
# replace na with row means
ind = which(is.na(List_null),arr.ind=T)
List_null[ind]=rowMeans(List_null,na.rm=T)[ind[,1]]
# replace na with zero
List_null[which(!complete.cases(List_null)),]=0

File: PythonDataAnalysis/Impute Missing Values.py
import pandas as pd
Data = pd.read_csv("Dataset.csv")

List_nulls = pd.DataFrame({'one':(1,3,8,6,9),
                           'two':(8,0,1,2,6)},
                            index = ['a','c','e','f','g'])
List_nulls

List_nulls2 = List_nulls.reindex(['a','b','c','d','e','f','g','h'])
List_nulls2

# pandas has a fillna function to "fill in" NA values with non-null data in a couple of ways
# 1. fill with zero
List_nulls2.fillna(0)
List_nulls2.fillna(999)

# 2. fill with the mean, column mean
List_nulls2.fillna(List_nulls2.mean())
List_nulls2.mean()

# 3. fill with the maximum number
List_nulls2.fillna(List_nulls2.max())

# 4. fill with the minimum number
List_nulls2.fillna(List_nulls2.min())


